### PR TITLE
WIP: Add 'OpenEXR' package

### PR DESCRIPTION
**Pull request by hjmallon**
* I've followed [this guide](https://hunter.readthedocs.io/en/latest/creating-new/create/cmake.html)
  step by step carefully. **[Yes]**

* I've tested this package remotely and have excluded all broken builds.
  Here is the links to the Travis/AppVeyor with status "All passed":

  * https://ci.appveyor.com/project/hjmallon/hunter/builds/31175845
  * https://travis-ci.org/hjmallon/hunter/builds/657217608

* I have submitted CI configs to https://github.com/cpp-pm/hunter-testing targeting `pkg.template` branch,
  see this merged pull request https://github.com/cpp-pm/hunter-testing/pull/51

* My change will resolve this "New package" request: https://github.com/cpp-pm/hunter/issues/128

---

Testing PR is not merged yet, so WIP.

### CODE REVIEW

File: cmake/configs/default.cmake [hunk 1]
```diff
@@ -78,7 +78,7 @@ hunter_default_version(HastyNoise VERSION 0.8.3)
 hunter_default_version(ICU VERSION 63.1-p5)
 hunter_default_version(IF97 VERSION 2.1.2)
 hunter_default_version(Igloo VERSION 1.1.1-hunter)
-hunter_default_version(IlmBase VERSION 2.4.0)
+hunter_default_version(IlmBase VERSION 2.4.0-p0)
```
Comment (rbsheth) [inline cmake/configs/default.cmake]: Should we make this `2.4.0-p1` so the OpenEXR and IlmBase packages are from the same archive?

Comment (hjmallon) [reply cmake/configs/default.cmake]: Makes sense, done

**Comment by hjmallon** [comment]
Here are some new builds with all 3 commits in (not finished at time of writing)
* https://travis-ci.org/hjmallon/hunter/builds/659099203
* https://ci.appveyor.com/project/hjmallon/hunter/builds/31288961

Also https://github.com/cpp-pm/hunter-testing/pull/51 needs merging before this one can be.

**Comment by rbsheth** [comment]
> Also [cpp-pm/hunter-testing#51](https://github.com/cpp-pm/hunter-testing/pull/51) needs merging before this one can be.

It's actually the other way. We need to merge this first and then the testing branch, as we need to update `pkg.template`.

**Comment by hjmallon** [comment]
> > Also [cpp-pm/hunter-testing#51](https://github.com/cpp-pm/hunter-testing/pull/51) needs merging before this one can be.
> 
> It's actually the other way. We need to merge this first and then the testing branch, as we need to update `pkg.template`.

Oh right, the text in the PR example says it should be merged I think

**Pull request merged**